### PR TITLE
Added toggle button

### PR DESCRIPTION
**Pull request by chelsi-001011**
Issue: #1

### CODE REVIEW

**Comment by chaitanya4vedi** [comment]
> Issue: #1

Ok Nice. But i think toggle button should toggle on off. Because now its just two static buttons. You can refer to [this](https://www.w3schools.com/howto/howto_css_switch.asp) to know how to do it.

**Comment by chelsi-001011** [comment]
> > Issue: #1
> 
> Ok Nice. But i think toggle button should toggle on off. Because now its just two static buttons. You can refer to [this](https://www.w3schools.com/howto/howto_css_switch.asp) to know how to do it.

@shahednasser  asked me to create to different buttons, but i can do this single button with on and off functionality?
Should i go ahead with it?

**Comment by chaitanya4vedi** [comment]
> > > Issue: #1
> > 
> > Ok Nice. But i think toggle button should toggle on off. Because now its just two static buttons. You can refer to [this](https://www.w3schools.com/howto/howto_css_switch.asp) to know how to do it.
> @chaitanya4vedi 
> @shahednasser  asked me to create two different static buttons, but I can do this single button with on and off functionality?
> Should I go ahead with it?

I see. Ok then. I approve it. For now I can't tell you if you should go ahead with that as I didn't read issue:#1. You can ask @shahednasser for that.

**Review by chaitanya4vedi — APPROVED** [review]

**Comment by chaitanya4vedi** [comment]
> > > Issue: #1
> > 
> > Ok Nice. But i think toggle button should toggle on off. Because now its just two static buttons. You can refer to [this](https://www.w3schools.com/howto/howto_css_switch.asp) to know how to do it.
> @chaitanya4vedi 
> @shahednasser  asked me to create two different static buttons, but I can do this single button with on and off functionality?
> Should I go ahead with it?

Also when you clone a branch then leave the `master` branch as it is and create a new branch with `git checkout -b new_branch_name` and then work on that new branch. Because sometimes the admin may ask you to revert some changes or make something as it was before. Or sometimes your changes may not be accepted. In that case it will be difficult for you to do that or proceed further. So it's better to have one branch i.e. `master` left as it is and then everytime you want to work on something just go to master with `git checkout master` and create a  new branch from there and then proceed. You can delete branches by `git branch -d branch_name`. And you can also delete branches remotely after it is merged or closed

**Review by shahednasser — CHANGES_REQUESTED** [review]
It all looks really great! I just have one thing to point out:

If someone is gonna use these buttons, they'll probably use them by changing the class of the button from off to on or the opposite. You're currently using `: before` for off btn and `:after` for on button.  So If I change from `toggle-off-btn` to `toggle-off-btn` the button's class and look will change but there will be no transition of the white button and the green background.

What I suggest is using `:before` (or after) to display the design of the white button generally on `toggle-btn`, and then make changes regarding position (left or right) based on if it's `toggle-off-btn` or `toggle-on-btn`. 

You should also add transition duration to `.toggle-btn` and `toggle-btn` before. This will make the transition clean. In order for the transition to look good, instead of using float right or left, you will need to set `left` or `right` and add position absolute to `:before` and position relative to `.toggle-btn`.

I hope what I am saying is clear. If you need any more explanation or any help please let me know!

**Comment by shahednasser** [comment]
Also just to clarify the buttons should stay static (not change on click). This is just for future usage of the button.

**Comment by chaitanya4vedi** [comment]
> Also just to clarify the buttons should stay static (not change on click). This is just for future usage of the button.

@shahednasser I have a question. We can add elevated buttons that press on click? Like a keyboard button. Or the button should be completely static like no movement on click.

**Comment by shahednasser** [comment]
> > Also just to clarify the buttons should stay static (not change on click). This is just for future usage of the button.
> 
> @shahednasser I have a question. We can add elevated buttons that press on click? Like a keyboard button. Or the button should be completely static like no movement on click.

Do you mean on click adding some action to the button?

**Comment by chaitanya4vedi** [comment]
> > > Also just to clarify the buttons should stay static (not change on click). This is just for future usage of the button.
> > 
> > @shahednasser I have a question. We can add elevated buttons that press on click? Like a keyboard button. Or the button should be completely static like no movement on click.
> 
> Do you mean on click adding some action to the button?

No action. Just transition. Like the transition of a key press

**Comment by shahednasser** [comment]
@chaitanya4vedi our project is just CSS though. Maybe we can add it to the website in index.js but I'm not sure if it will cause confusion as to where the actual button being used on click will toggle on/off

**Comment by chaitanya4vedi** [comment]
> @chaitanya4vedi our project is just CSS though. Maybe we can add it to the website in index.js but I'm not sure if it will cause confusion as to where the actual button being used on click will toggle on/off

No JavaScript or jQuery. I mean with just CSS.

**Comment by shahednasser** [comment]
Ok @chaitanya4vedi I think let's implement them statically with the changes requested above and once it is merged you can work on that if you want (because I'm not sure I completely understand what you mean)

File: assets/js/buttons-examples.js [hunk 1]
```diff
@@ -352,6 +352,23 @@ $(document).ready(function () {
...
+            {
+              classes: "toggle-on-btn",
+              text: "",
+            },
+            {
+              classes: "toggle-off-btn",
+              text: "",
+            },
+          ],
+          shouldHaveRoundedType: false,
```
[the remark this answers is not in the document]
Comment (chelsi-001011) [reply assets/js/buttons-examples.js:369]: @shahednasser  I think I got your point you want me to make transitions to reach "on" and "off" positions and a static design to toggle-btn?

[the remark this answers is not in the document]
Comment (shahednasser) [reply assets/js/buttons-examples.js:369]: yes @chelsi-001011 I want you to make it possible to make transition seamless between the two states just by switching the classes of on and off. It shouldn't be portrayed in the website for now (we're still discussing the best way to display it), it just should be possible. Keep the buttons static.

File: assets/js/buttons-examples.js [hunk 2]
```diff
@@ -415,6 +415,23 @@ $(document).ready(function () {
...
+            {
+              classes: "toggle-on-btn",
+              text: "",
+            },
+            {
+              classes: "toggle-off-btn",
+              text: "",
+            },
+          ],
+        }
```
Comment (shahednasser) [inline assets/js/buttons-examples.js:434]: There's a syntax error here please add a comma

**Review by shahednasser — CHANGES_REQUESTED** [review]

File: assets/js/buttons-examples.js (hunk 2 above)
Comment (chelsi-001011) [reply assets/js/buttons-examples.js]: @shahednasser  I m in the middle of a really bad merge conflict situation. Should I just create a new pull request?

File: src/components/special/_toggle.less [hunk 3]
```diff
@@ -0,0 +1,38 @@
+.toggle-btn {
+  width: 90px;
+  height: 50px;
+  border-radius: 50px;
+  border: 0;
+  transition: all 0.5s ease;
+  outline: none;
+  display: flex;
```
Comment (shahednasser) [inline src/components/special/_toggle.less:8]: Change display to `inline-flex`

**Review by shahednasser — CHANGES_REQUESTED** [review]

File: src/components/special/_toggle.less (hunk 3 above)
Comment (chelsi-001011) [reply src/components/special/_toggle.less:8]: @shahednasser should I close this pull request and create a new one?

Comment (shahednasser) [reply src/components/special/_toggle.less:8]: Sure if there are too many conflicts then do that

Comment (chelsi-001011) [reply src/components/special/_toggle.less:8]: okay! Ill keep in mind these changes that u have suggested and create a new pr

**Comment by shahednasser** [comment]
Also @chelsi-001011 please place the button in the correct Alphabetical order under "Special"

**Pull request closed without merging**